pkg/iacinventory/github_repo.go

```diff
@@ -0,0 +1,184 @@
+package iacinventory
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/soluble-ai/soluble-cli/pkg/log"
+)
+
+type GithubRepo struct {
```

Comment:
This seems to duplicate the Repo type?

Reply:
Kind of. GithubRepo is the data we can/do extract for GitHub. I explicitly separated the GitHub type from the generic `Repo` type. I manually construct the `Repo` from the `GithubRepo`. Rationale is that other providers (GitLab, Bitbucket, etc) down the road may not, for whatever reason, allow pulling out the requisite data. Or, the "name" of each may differ (`full_name` vs `name`), etc.

Reply:
Yeah, wouldn't worry about that.  If Repo is some denormalized structure with everything we could possibly send that's fine.

Reply:
I created an Repo interface with the basic methods that are needed for any given repo for data extraction. This is probably what I should have done to begin with. Can add a `MarshalJSON` function that marshals the data to JSON as needed.

Reply:
Reverted that change. Just using the `GithubRepo` type for now. Will deal with the interface after we hook up another provider.